spec/integration/media_convert_adapter_spec.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+require 'spec_helper'
+require 'aws-sdk-cloudwatchevents'
+require 'aws-sdk-cloudwatchlogs'
+require 'aws-sdk-mediaconvert'
+require 'aws-sdk-s3'
+require 'json'
+require 'active_encode/spec/shared_specs'
+require 'active_support/json'
+require 'active_support/time'
+
+def reconstitute_response(fixture_path)
+  HashWithIndifferentAccess.new(ActiveSupport::JSON.decode(File.read(File.join("spec/fixtures", fixture_path))))
+end
+
+describe ActiveEncode::EngineAdapters::MediaConvertAdapter do
+  around do |example|
+    ActiveSupport.parse_json_times = true
+    Time.zone = 'America/Chicago'
+
+    # Setting this before each test works around a stubbing + memoization limitation
+    ActiveEncode::Base.engine_adapter = :media_convert
+    ActiveEncode::Base.engine_adapter.role = 'arn:aws:iam::123456789012:role/service-role/MediaConvert_Default_Role'
+    ActiveEncode::Base.engine_adapter.output_bucket = 'output-bucket'
+    example.run
+    ActiveEncode::Base.engine_adapter = :test
+  end
+
+  let(:job_id) { "1625859001514-vvqfwj" }
+  let(:mediaconvert) { Aws::MediaConvert::Client.new(stub_responses: true) }
+  let(:cloudwatch_events) { Aws::CloudWatchEvents::Client.new(stub_responses: true) }
+  let(:cloudwatch_logs) { Aws::CloudWatchLogs::Client.new(stub_responses: true) }
+
+  before do
+    mediaconvert.stub_responses(:describe_endpoints, reconstitute_response("media_convert/endpoints.json"))
+
+    allow(Aws::MediaConvert::Client).to receive(:new).and_return(mediaconvert)
+    allow(Aws::CloudWatchEvents::Client).to receive(:new).and_return(cloudwatch_events)
+    allow(Aws::CloudWatchLogs::Client).to receive(:new).and_return(cloudwatch_logs)
+  end
+
+  let(:created_job) do
+    mediaconvert.stub_responses(:create_job, reconstitute_response("media_convert/job_created.json"))
+
+    ActiveEncode::Base.create(
+      "s3://input-bucket/test_files/source_file.mp4",
+      output_prefix: "active-encode-test/output",
+      outputs: [
+        { preset: "System-Avc_16x9_1080p_29_97fps_8500kbps", modifier: "-1080" },
+        { preset: "System-Avc_16x9_720p_29_97fps_5000kbps", modifier: "-720" },
+        { preset: "System-Avc_16x9_540p_29_97fps_3500kbps", modifier: "-540" }
+      ],
+      use_original_url: true
+    )
+  end
+
+  let(:running_job) do
+    mediaconvert.stub_responses(:get_job, reconstitute_response("media_convert/job_progressing.json"))
+    ActiveEncode::Base.find(job_id)
+  end
+
+  let(:canceled_job) do
+    mediaconvert.stub_responses(:get_job, reconstitute_response("media_convert/job_canceled.json"))
+    ActiveEncode::Base.find(job_id)
+  end
+
+  let(:cancelling_job) do
+    mediaconvert.stub_responses(:get_job, reconstitute_response("media_convert/job_canceled.json"))
```

Comment:
Is this supposed to be `job_canceling.json`?
```suggestion
    mediaconvert.stub_responses(:get_job, reconstitute_response("media_convert/job_canceling.json"))
```

Reply:
The test needs them both. Fixed.